src/api/v1/Auth/controller.js

```diff
@@ -1,22 +1,137 @@
 const {
     sendSuccessResp,
     sendFailureResp,
-} = require('../../../utils/response.js');
+} = require("../../../utils/response.js");
+
+const { User, Admin } = require("../../../database/index.js");
+
+const { createToken } = require("../../../services/jwt.js");
+
+const { hash, compareHash } = require("../../../services/encryption.js");
+
+const roleArray = [User, Admin];
 
 module.exports = {
-    signup: (req, res) => {
-        sendSuccessResp(res,{
-            data: {
-                message: "Mann Lo Ki User Sign UP ho gya"
+    signup: async (req, res) => {
+        const { role, name, mobile, email, password } = req.body;
+        const model = roleArray[role];
+        try {
+            const data = await findUser(role, email);
+            const hashedPassword = await hash(password);
+
+            if (!data) {
+                const { dataValues } = await model.create({
+                    name: name,
+                    email: email,
+                    mobile: mobile,
+                    password: hashedPassword,
+                });
+
+                if (Object.keys(dataValues)?.length) {
+                    const token = await generateUserToken(dataValues, role);
+
+                    return sendSuccessResp(res, {
+                        status: 200,
+                        data: {
+                            isExist: 0,
+                            token,
+                            message: "Sign Up Successfully!",
+                        },
+                    });
+                }
+            } else {
+                return sendFailureResp(res, {
+                    status: 400,
+                    data: {
+                        isExist: 1,
+                        message: "User already exists",
+                    },
+                });
             }
-        })
+        } catch (err) {
+            return sendFailureResp(res, {
+                status: 500,
+                data: {
+                    message: "Something went wrong!",
+                },
+            });
+        }
     },
 
```

Comment:
dont just send the token in body , aslo set it as cookie

Reply:
done